cfgov/jinja2/v1/_includes/organisms/mega-menu.html

```diff
@@ -1,61 +1,146 @@
+{% import '_vars-mega-menu.html' as vars with context %}
+{% set base_class = 'o-mega-menu' %}
 
 {# ==========================================================================
 
-   sub_nav()
+   _classes()
 
    ==========================================================================
 
    Description:
 
-   Creates a mega menu sub menu content markup when given:
+   Creates mega menu CSS classes when given:
+
+   nav_depth:    Level of menu nesting. 1 equals the root menu.
+
+   suffix:       Suffix to add to the classes.
+                 Default is ''.
+
+   ========================================================================== #}
+
+{%- macro _classes( nav_depth, suffix='' ) -%}
+{%- set general_class = base_class ~ '_content' -%}
+{%- set depth_class = base_class ~ '_content-' ~ nav_depth -%}
+{{ general_class ~ suffix ~ ' ' ~ depth_class ~ suffix }}
+{%- endmacro -%}
+
+
+{# ==========================================================================
+
+   _nav_list()
 
-   sub_nav_title:   A string used for the sub nav title.
+   ==========================================================================
 
-   sub_nav_landing: A string used for the url of the sub nav landing page.
+   Description:
 
-   sub_nav_items:   An array of tuples.
+   Creates a mega menu sub menu list markup when given:
 
-   sub_nav_media:   A string composed of html markup.
+   nav_depth: Level of menu nesting. 1 equals the root menu.
+
+   nav_items: An array of tuples containing menu link info.
 
    ========================================================================== #}
 
-{% macro _sub_nav(sub_nav_title, sub_nav_landing, sub_nav_items, sub_nav_media) %}
-<div class="o-mega-menu_subcontent" aria-expanded="false">
-    <div class="o-mega-menu_subcontent-wrapper">
+{% macro _nav_list( nav_depth, nav_items ) %}
+{%- for nav_group in nav_items %}
+<ul class="list
+           list__unstyled
+           {{ _classes( nav_depth, '-list' ) }}">
+    {% if nav_depth == 1 -%}
+    <li class="list_item
+               {{ _classes( nav_depth, '-item' ) }}">
+        {% import 'molecules/global-header-cta.html' as global_header_cta with context %}
+        {{ global_header_cta.render() }}
+    </li>
+    {%- endif %}
+    {%- for nav_caption,
+            nav_url,
+            nav_children in nav_group %}
+    <li class="list_item
+               {{ _classes( nav_depth, '-item' ) }}"
+        {{ 'data-js-hook=flyout-menu' if nav_children | count > 0 else '' }}>
+        {# TODO: Remove nav_depth < 2 check on has-children class when 3rd level transitions are in. #}
+        <a class="{{ 'u-link__disabled' if nav_url == '' else '' }}
```

Comment:
This is a lot of logic to place in the template. Would be nice if some of this could be ported to Django